reactor-netty-core/src/main/java/reactor/netty/resources/LoopResources.java

```diff
@@ -175,6 +175,18 @@ static LoopResources create(String prefix, int selectCount, int workerCount, boo
 		return new DefaultLoopResources(prefix, selectCount, workerCount, daemon, colocate);
 	}
 
+	/**
+	 * Return a {@link LoopResources} that will share the same {@link EventLoopGroup} from this {@link LoopResources}, but with
+	 * colocation support disabled (if it was enabled).<p>
+	 * Disposing the returned {@link LoopResources} will also dispose the {@link EventLoopGroup} from the origin {@link LoopResources}.
+	 *
+	 * @return a LoopResources that will share the same {@link EventLoopGroup} from this {@link LoopResources}
+	 * @since 1.1.9
+	 */
+	default LoopResources disableColocation() {
```

Comment:
Does it make sense to have `disableColocation` on `TcpResources` as well?

Reply:
The LoopResources.disableColocation is generic and can be applied to any default LoopResources, including the TcpResources (which implements the LoopResources interface), or the default HttpResources, or even the default UdpResources.

For example, in Mariadb driver, if users don't have provided any custom LoopResources, you can then configure the driver with a mutated version of the default TcpResources with colocation disabled like this in [MariadbConnectionConfiguration](https://github.com/mariadb-corporation/mariadb-connector-r2dbc/blob/master/src/main/java/org/mariadb/r2dbc/MariadbConnectionConfiguration.java#L142):

```
    this.loopResources = loopResources != null ? loopResources : TcpResources.get().disableColocation();

```

The TcpResources inherits from the LoopResources.
it's important to note that you need to use the mutated LoopResources that is **returned** by the disableColocation() method (which does not alter the original LoopResources that remains unchanged and  stay colocated mode, so other components which are happy with this default mode won't be impacted, like the HttpClient).

If you want to share the default HttpResources with the R2DBC driver, you could also do it (in this case R2DBC will reuse the same event loop group from Http applications:

```
    this.loopResources = loopResources != null ? loopResources : HttpResources.get().disableColocation();

```

all this to say that the `LoopResources.disableColocation()` is generic and applies to either one of the existing default loop resources (`HttpResources.get()`, `TcpResources.get()`, or `UdpResources.get())`, or to even any custom colocated LoopResources created using one of the `LoopResources.create(...)` methods.